File: examples/equos-python-integration/src/python_integration/create_face.py
import sys

from equos.client.errors import UnexpectedStatus
from equos.models import CreateEquosFaceRequest, CreateEquosFaceRequestIdentity

from python_integration.client import get_client


def create_face() -> None:
    client = get_client()
    identity = CreateEquosFaceRequestIdentity.TOMMY

    try:
        face = client.faces.create(CreateEquosFaceRequest(identity=identity))
        print("Face created:", face.id)
    except UnexpectedStatus as err:
        if err.status_code != 409:
            raise
        existing = client.faces.list(take=50)
        match = next(
            (f for f in existing.faces if f.identity.value == identity.value),
            None,
        )
        print(
            f'A face with identity "{identity.value}" already exists for this organization.',
            file=sys.stderr,
        )
        if match is not None:
            print(f"→ Reuse id: {match.id}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    create_face()

File: examples/equos-nodejs-integration/src/create-face.ts
import {
  CreateEquosFaceRequestIdentityEnum,
  ResponseError,
} from "@equos/node-sdk";
import { client } from "./client.js";

// `identity` picks a face from the Equos Gallery.
try {
  const face = await client.faces.createFace({
    createEquosFaceRequest: {
      identity: CreateEquosFaceRequestIdentityEnum.Tommy,
    },
  });
  console.log("Face created:", face.id);
  console.log(face);
} catch (err) {
  if (err instanceof ResponseError && err.response.status === 409) {
    const existing = await client.faces.listFaces({ take: 50 });
    const match = existing.faces.find(
      (f) => f.identity === CreateEquosFaceRequestIdentityEnum.Tommy
    );
    console.error(
      `A face with identity "${CreateEquosFaceRequestIdentityEnum.Tommy}" already exists for this organization.`
    );
    if (match) console.error(`→ Reuse id: ${match.id}`);
    process.exit(1);
  }
  throw err;
}
